Cargo.toml

```diff
@@ -25,7 +25,9 @@ at-commands = "0.5.2"
 no-std-net = "0.6.0"
 critical-section = "1.1"
 embedded-io = { version = "0.4.0", features = ["async"] }
+heapless = { version = "0.7.16", optional = true }
```

Comment:
Arrayvec is already included which also has an Array string type. Unless you need something specific from the types that headless has, you can use the ArrayVec crate